app/models/hathi_links.rb

```diff
@@ -1,54 +1,67 @@
 # frozen_string_literal: true
 
-class HathiLinks
-  attr_reader :document
+module HathiLinks
+  def hathi_links
+    return nil unless has_hathi_links?
 
-  def initialize(document)
-    @document = document
-    @ht_hash = ht_hash
+    {
+      text: ht_link_text,
+      url: ht_url,
+      additional_text: ht_additional_text
+    }
   end
 
-  def ht_hash
-    JSON.parse(document['hathitrust_struct']&.first)
-  end
+  private
 
-  def id
-    ht_hash['ht_id']
-  end
+    def has_hathi_links?
+      @_source['hathitrust_struct'].present?
+    end
 
-  def bib_key
-    ht_hash['ht_bib_key']
-  end
+    def hathi_links_field
+      @_source['hathitrust_struct']
+    end
 
-  def access
-    ht_hash['access']
-  end
+    def ht_hash
+      JSON.parse(hathi_links_field&.first)
+    end
 
-  def url
-    url = if id.present?
-            I18n.t('blackcat.hathitrust.mono_url') + id
-          else
-            I18n.t('blackcat.hathitrust.multi_url') + bib_key
-          end
+    def ht_id
+      ht_hash['ht_id']
+    end
 
-    url += I18n.t('blackcat.hathitrust.url_append') if etas_item?
-    url
-  end
+    def ht_bib_key
+      ht_hash['ht_bib_key']
+    end
 
-  def link_text
-    text = I18n.t('blackcat.hathitrust.public_domain_text') if access == 'allow'
-    text = I18n.t('blackcat.hathitrust.restricted_access_text') if access == 'deny' && !Settings&.hathi_etas
-    text = I18n.t('blackcat.hathitrust.etas_text') if etas_item?
-    text
-  end
+    def ht_access
+      ht_hash['access']
+    end
 
-  def additional_text
-    return unless etas_item?
+    def ht_url
+      url = if ht_id.present?
+              I18n.t('blackcat.hathitrust.mono_url') + ht_id
+            else
+              I18n.t('blackcat.hathitrust.multi_url') + ht_bib_key
+            end
 
-    I18n.t('blackcat.hathitrust.etas_additional_text')
-  end
+      url += I18n.t('blackcat.hathitrust.url_append') if etas_item?
+      url
+    end
 
-  def etas_item?
-    access == 'deny' && Settings&.hathi_etas
-  end
+    def ht_link_text
+      I18n.t('blackcat.hathitrust.public_domain_text') if ht_access == 'allow'
+      I18n.t('blackcat.hathitrust.restricted_access_text') if ht_access == 'deny' && !Settings&.hathi_etas
+
+      I18n.t('blackcat.hathitrust.etas_text')
```

Comment:
would not this always return etas_text? 🤔

Reply:
Yep. Did that one too quick